presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1467,6 +1470,16 @@ public Expression rewriteIdentifier(Identifier reference, Void context, Expressi
         {
             List<Set<Expression>> computedGroupingSets = ImmutableList.of(); // empty list = no aggregations
 
+            boolean isGroupingOperationPresent = outputExpressions.stream()
+                    .anyMatch(GroupingOperationAnalyzer::containsGroupingOperation);
+
+            if (isGroupingOperationPresent && !node.getGroupBy().isPresent()) {
```

Comment:
Yep you're right. Moved this check to a separate function called `analyzeGroupingFunctions()`. I prefer that name over your suggestion because `analyzeGroupings()` is a little more ambiguous and can be confused with the analysis of other aggregates.